src/libs/antares/study/parts/short-term-storage/cluster.cpp

```diff
@@ -63,17 +66,19 @@ bool STStorageCluster::loadFromSection(const IniFile::Section& section)
     return true;
 }
 
-bool STStorageCluster::validate()
-{
-    logs.debug() << "Validating properties and series for st storage: " << id;
-    return properties.validate() && series->validate();
-}
-
 bool STStorageCluster::loadSeries(const std::string& folder) const
 {
     bool ret = series->loadFromFolder(folder);
     series->fillDefaultSeriesIfEmpty(); // fill series if no file series
     return ret;
 }
 
+bool STStorageCluster::validate(bool simplexIsWeek, unsigned int start, unsigned int end)
+{
+    logs.debug() << "Validating properties and series for st storage: " << id;
+    return properties.validate(simplexIsWeek) && series->validate() &&
+        series->validateInitialLevelSimplex(simplexIsWeek, properties.initialLevel,
+            properties.storagecycle.value(), start, end);
```

Comment:
storagecycle may not have a value at this stage (in unit tests for example). Pass it (as an std::optional) instead of its value ?

Reply:
This property is mandatory in properties.cpp:validate(), thus cannot fail